rust/agents/validator/src/submit.rs

```diff
@@ -69,7 +69,7 @@ impl ValidatorSubmitter {
     }
 
     async fn main_task(self) -> Result<()> {
-        let reorg_period = Some(self.reorg_period);
+        let reorg_period = if self.reorg_period == 0 { None } else { Some(self.reorg_period) };
```

Comment:
definitely have no idea how this works but is this not the same as
```suggestion
        let reorg_period = self.reorg_period;
```
?